zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/usertask/UserTaskRecord.java

```diff
@@ -103,8 +103,42 @@ public final class UserTaskRecord extends UnifiedRecordValue implements UserTask
   private final LongProperty elementInstanceKeyProp = new LongProperty("elementInstanceKey", -1L);
   private final StringProperty tenantIdProp =
       new StringProperty("tenantId", TenantOwned.DEFAULT_TENANT_IDENTIFIER);
+
+  /**
+   * Tracks the names of user task attributes that have been modified.
+   *
+   * <p>This property was introduced to enable tracking of attributes that were truly modified.
+   * Since our `msgpack` implementation doesn't support `null` values, all record attributes have
+   * default non-null values. As a result, without `changedAttributesProp`, it would be impossible
+   * to determine whether an attribute was explicitly modified or simply retained its default value.
+   * This property solves that issue by explicitly listing attributes that were changed.
+   *
+   * <p>The content of `changedAttributesProp` field depend on the type of event being processed:
```

Comment:
🔧 It's not just events but also for commands, so any record.
```suggestion
   * <p>The content of `changedAttributesProp` field depend on the type of record:
```